genesyscloud/resource_genesyscloud_knowledge_category_test.go

```diff
@@ -88,20 +173,36 @@ func generateKnowledgeCategoryResource(resourceName string, knowledgeBaseResourc
         }
         `, resourceName,
 		knowledgeBaseResource,
-		generateKnowledgeCategoryRequestBody(categoryName, categoryDescription),
+		generateKnowledgeCategoryRequestBody(categoryName, categoryDescription, ""),
+	)
+	return category
+}
+
+func generateKnowledgeCategoryChildResource(resourceName string, knowledgeBaseResource string, categoryName string, categoryDescription string, parentCategoryName string) string {
+	category := fmt.Sprintf(`
+	resource "genesyscloud_knowledge_category" "%s" {
+		knowledge_base_id = genesyscloud_knowledge_knowledgebase.%s.id
+		%s
+	}
+	`, resourceName,
+		knowledgeBaseResource,
+		generateKnowledgeCategoryRequestBody(categoryName, categoryDescription,
+			"parent_id = genesyscloud_knowledge_category."+parentCategoryName+".id"),
 	)
 	return category
 }
 
-func generateKnowledgeCategoryRequestBody(categoryName string, categoryDescription string) string {
+func generateKnowledgeCategoryRequestBody(categoryName string, categoryDescription string, parrentCategoryName string) string {
 
 	return fmt.Sprintf(`
         knowledge_category {
             name = "%s"
             description = "%s"
+			%s
```

Comment:
I would leave this as 

```
parent_id = %s
```

and pass `genesyscloud_knowledge_category."+parentCategoryName+".id"` into the function. And if you do not want to provide a value, you can pass `util.NullValue` as a parameter. This is a constant value for "null" , and it is the same as omitting the field from the resource altogether.

If you want the resource to look like this

```
parent_id = "some hard-coded string"
```

You would pass `strconv.Quote("some hard-coded string")` into the function to add the quotation marks

Reply:
thanks for the suggestions